remotes/docker/pusher.go

```diff
@@ -382,6 +384,7 @@ func (pw *pushWriter) Commit(ctx context.Context, size int64, expected digest.Di
 	if resp.err != nil {
 		return resp.err
 	}
+	resp.Body.Close()
```

Comment:
Note that `default:` below calls `remoteserrors.NewUnexpectedStatusErr`, which uses the response Body, so wondering if a `defer` would work here?


For consistency with the code below, perhaps explicitly reference using `resp.Response.Body` instead of `resp.Body`

```suggestion
	defer resp.Response.Body.Close()
```

Reply:
Fixed as suggested, they are definitely better.  Thanks for the comments!